DEVELOPING.md

```diff
@@ -1,75 +1,205 @@
+The unit tests should all succeed.
+The compiled `tla2tools.jar` will be output to `tlatools/org.lamport.tlatools/dist/tla2tools.jar`, and can be used the same as any `tla2tools.jar` you download from the releases.
 
-Make sure that you're using Java 11+. If you're using a lower version you'll see compilation errors.
+Build & Test Toolbox IDE
+------------------------
+Install the following dependencies to your path:
+ * [Java Development Kit](https://adoptium.net/) version 11+
+ * [Apache Ant](https://ant.apache.org/) version 1.9.8+
+ * [Apache Maven](https://maven.apache.org/) version 3.9.7+
 
-You can build TLC from a fresh clone of this repository+:
+Clone this repo & open a shell in its root, then run:
 ```bash
-$ git clone https://github.com/tlaplus/tlaplus.git
-$ cd tlaplus/tlatools/org.lamport.tlatools
-$ ant -f customBuild.xml info clean compile compile-test dist
+mvn install -Dmaven.test.skip=true # Builds Toolbox
+mvn verify # Runs tests
 ```
 
-The `tlatools.jar` will be created in the `org.lamport.tlatools/dist` directory.
+The Toolbox distributables will be output in the `toolbox/org.lamport.tla.toolbox.product.product/target/products` directory.
+
+Developing with the Eclipse IDE
+-------------------------------
+You are free to use your own editor, but this project is largely developed in the [Eclipse](https://www.eclipse.org/) integrated development environment (IDE).
+
+### Developing the TLA⁺ Tools in Eclipse
+
+Developing the TLA⁺ Tools in Eclipse requires the [Eclipse IDE for RCP and RAP Developers](https://www.eclipse.org/downloads/packages/release/2024-06/r/eclipse-ide-rcp-and-rap-developers) edition, or any other edition that comes with the OSGi Plugin Development Extension.
+After you have installed Eclipse, use it open the `tlatools/org.lamport.tlatools` project directory.
+You should then be able to build, run, test, and debug the TLA⁺ Tools.
+
+When running unit tests in Eclipse, many of the tests depend on TLA⁺ module and model files in the repository.
+These tests need to know where to look for these files.
+For this reason, you will often need to modify the test run configurations by adding `-Dbasedir=${workspace_loc:tlatools}` to the VM arguments (not the program arguments) in the Arguments tab of the run configurations menu.
+You should then be able to run & debug JUnit tests in the Eclipse UI.
 
-Test that the build was successful by using an example model:
+Note that running build or test operations with Ant can occasionally interfere with Eclipse's build system and cause Eclipse to report nonexistent compilation errors in the UI.
+For this reason you should run Ant from within Eclipse by opening Window > Show View > Other > Ant > Ant, then adding the `customBuild.xml` file and running selected targets.
```

Comment:
The "RunAllTLCTests" should show up under the run external tool button in Eclipse automatically.

<img width="357" alt="image" src="https://github.com/tlaplus/tlaplus/assets/88777/da696f39-1bf7-4515-825b-7c4355fb90d2">

Reply:
Coming from tlatools/org.lamport.tlatools/ide/RunAllTLCTests.launch